var/spack/repos/builtin/packages/mumps/package.py

```diff
@@ -141,4 +163,19 @@ def install(self, spec, prefix):
         install_tree('lib', prefix.lib)
         install_tree('include', prefix.include)
         if '~mpi' in spec:
-            install('libseq/libmpiseq.a', prefix.lib)
+            lib_dsuffix = '.dylib' if sys.platform == 'darwin' else '.so'
+            lib_suffix = lib_dsuffix if '+shared' in spec else '.a'
+            install('libseq/libmpiseq%s' % lib_suffix, prefix.lib)
+
+        if '~mpi' in spec: # FIXME: A process or daemon was unable to complete a TCP connection to another process
+            with working_dir('examples', create=False):
+                mpirun = join_path(self.spec['mpi'].prefix.bin, 'mpirun') # FIXME: avoid hardcoding mpirun (could be mpiexec, mpiexec.hydra, etc)
```

Comment:
I guess we are not talking of the same thing.
I just added a `print spec['mpi'].prefix` in the `install` function
`spack install mumps~mpi` results in 

```
Traceback (most recent call last):
  File "/home/richart/spack/lib/spack/spack/build_environment.py", line 367, in fork
    function()
  File "/home/richart/spack/lib/spack/spack/package.py", line 910, in build_process
    self.install(self.spec, self.prefix)
  File "/home/richart/spack/var/spack/repos/builtin/packages/mumps/package.py", line 124, in install
    print spec['mpi'].prefix
  File "/home/richart/spack/lib/spack/spack/spec.py", line 1571, in __getitem__
    raise KeyError("No spec with name %s in %s" % (name, self))
KeyError: 'No spec with name mpi in mumps@5.0.1%gcc@4.8.5+complex+double+float~idx64~metis~mpi~parmetis~ptscotch~scotch=linux-x86_64^openblas@0.2.17%gcc@4.8.5=linux-x86_64'
```